src/elements/feedback.html

```diff
@@ -0,0 +1,289 @@
+<link rel="import" href="../../bower_components/polymer/polymer-element.html">
+<link rel="import" href="../../bower_components/paper-input/paper-textarea.html">
+<link rel="import" href="../../bower_components/star-rating/star-rating.html">
+<link rel="import" href="../../bower_components/paper-button/paper-button.html">
+<link rel="import" href="../mixins/redux-mixin.html">
+
+<dom-module id="feedback-block">
+    <template>
+      <style>
+        #feedback-comment {
+          width: 100%;
+        }
+
+        .star-rating, .caption {
+          display: inline-block;
+          vertical-align: bottom;
+        }
+
+        paper-button {
+          margin: 0;
+          line-height: 1.4;
+          border: 1.5px solid var(--default-primary-color);
+        }
+
+        paper-button[primary] {
+          background-color: var(--default-primary-color);
+          color: var(--text-primary-color);
+        }
+
+        paper-button.delete-button {
+          color: var(--text-accent-color);
+          padding: -2px;
+        }
+      </style>
+
+      <div class="container">
+        <div class="caption">{$ feedback.voteCaption $}:</div>
+        <star-rating class="star-rating" rating="{{rating}}"></star-rating>
+
+        <paper-textarea
+          id="commentInput"
+          hidden$="[[!rating]]"
+          label="Leave a comment here (optional)"
+          no-label-float
+          value="{{comment}}"
+        ></paper-textarea>
+        <paper-button
+          primary
+          hidden$="[[!rating]]"
+          on-tap="_sendFeedback"
+          ga-on="click"
+          ga-event-category="feedback"
+          ga-event-action="send feedback"
+          ga-event-label$="submit the [[rating]] stars feedback"
+        >
+          [[buttonLabel]]
+        </paper-button>
+        <paper-button
+          class="delete-button"
+          hidden$="[[!showDeleteButton]]"
+          on-tap="_dispatchDeleteFeedback"
+          ga-on="click"
+          ga-event-category="feedback"
+          ga-event-action="delete feedback"
+          ga-event-label$="delete the feedback record"
+        >
+          {$ feedback.deleteFeedback $}
+        </paper-button>
+      </div>
+    </template>
+
+    <script>
+      class Feedback extends ReduxMixin(Polymer.Element) {
```

Comment:
There is a lot of logic and methods in this class, I wonder if there might be some simplification that could happen.

Reply:
If we think about it, the module actually handles a lot of stuff..
- Makes sure the user is authenticated and checks if it is appropriate to display the feedback box;
- Fetches previous feedback;
- Maintains the fetched data in the redux state;
- Updates properly all info when the user switches between sessions;
- Submits the feedback to the db;